addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3045,6 +3061,9 @@ var SnippetsMenu = Widget.extend({
         } else {
             $toolbarContainer.show();
         }
+
+        const isInsideTD = !!(range && $(range.startContainer).parents('.o_editable td') && $(range.endContainer).parents('.o_editable td'));
+        $toolbarTableContainer.toggle(isInsideTD);
```

Comment:
In general we prefer not to "toggle" stuff but to toggle a d-none class or something like that. Unless fixed, toggle only restore display: block, not the original display.